test/cppsuite/src/main/thread_worker.cpp

```diff
@@ -210,6 +210,47 @@ thread_worker::remove(scoped_cursor &cursor, uint64_t collection_id, const std::
     return (ret == 0);
 }
 
+bool
+thread_worker::truncate(uint64_t collection_id, std::optional<std::string> start_key,
+  std::optional<std::string> stop_key, const std::string &config)
+{
+    WT_DECL_RET;
+
+    wt_timestamp_t ts = tsm->get_next_ts();
+    ret = txn.set_commit_timestamp(ts);
+    testutil_assert(ret == 0 || ret == EINVAL);
+    if (ret != 0) {
+        txn.set_needs_rollback(true);
+        return (false);
+    }
+
+    const std::string coll_name = database::build_collection_name(collection_id).c_str();
```

Comment:
Build collection name is generally used for creating new collections as it gets the standard format of a collection name. Here I would suggest calling `get_collection` that way we confirm the collection exists prior to opening the cursor.

```
const std::string = db.get_collection(id).name;
```
Side note: calling `c_str()` here is a bit odd as you're taking a std::string, converting it to char * then converting it back to std::string.

Reply:
Thanks for clarifying that, can definitely change the collection name stuff. As for the `c_str()` thing, I originally didn't have this conversion but I don't think the set key will work properly if we don't use `c_str()`. I wrote a basic test locally to insert some keys, truncate, and then check if they exist, and looks like we do need the conversion for it to work.

Reply:
Yeah the call into set_key will need the c_str() I was just noting that I don't think it helped when assigning to std::string.

Reply:
Yup I see now. I just made the `coll_name` variable as we're using the collection name multiple times.